node_modules/oae-principals/lib/api.group.js

```diff
@@ -208,39 +217,60 @@ var getGroupMembers = module.exports.getGroupMembers = function(ctx, groupId, st
             return callback({'code': 404, 'msg': util.format('Couldn\'t find principal: %s', groupId)});
         }
 
-        // Ensure proper permissions
-        canViewMembers(ctx, group, function(err, canViewMembers) {
+        // Get the members library to which the current user has access
+        return _getMembersLibrary(ctx, group, null, start, limit, callback);
+    });
+};
+
+/**
+ * Get the members library of the given group
+ *
+ * @param  {Context}            ctx                     Standard context object containing the current user and the current tenant
+ * @param  {Group}              group                   The group for which to get the members library
+ * @param  {Boolean}            [hasRole]               Whether or not it has already been determined that the current user has explicit access to the group. If truthy, it implies they do. If falsey, it implies that we don't know yet
+ * @param  {String}             [start]                 The group paging token from which to start fetching group members. Defaults to the beginning of the library
+ * @param  {Number}             [limit]                 The maximum number of results to return. Default: 10
+ * @param  {Function}           callback                Standard callback function
+ * @param  {Object}             callback.err            An error that occurred, if any
+ * @param  {User[]|Group[]}     callback.members        An array of the direct members of the group
+ * @param  {String}             callback.nextToken      The value to provide in the `start` parameter to get the next set of results
+ * @api private
+ */
+var _getMembersLibrary = function(ctx, group, hasRole, start, limit, callback) {
+    // Ensure proper permissions if we haven't determined that they have an explicit role
+    OaeUtil.invokeIfNecessary(!hasRole, LibraryAPI.Authz.resolveTargetLibraryAccess, ctx, group.id, group, function(err, hasAccess, visibility) {
+        if (err) {
+            return callback(err);
+        } else if (hasRole) {
```

Comment:
It does, but `hasRole` is much cheaper than `getRole` which `resolveTargetLibraryAccess` does. It's just a bit of a performance-related short-circuit to pass in the `hasRole`, because there is one context where we have already determined that the user has a role, which gives us enough information to proceed without getting their actual role.